internal/events/Database.go

```diff
@@ -0,0 +1,353 @@
+package events
+
+import (
+	"context"
+	"crypto/md5"
+	"database/sql"
+	"encoding/hex"
+	"time"
+
+	_ "github.com/jackc/pgx/v5/stdlib"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+
+	"GopherMart2/internal/errorsgm"
+)
+
+type OperationDB struct {
+	OrderNumber string
+	Login       string
+	UploadedAt  string
+	Status      string
+	Operation   string
+	Points      int
+}
+
+type UserDB struct {
+	Login           string
+	Password        string
+	CurrentPoints   int
+	WithdrawnPoints int
+}
+
+func (OperationDB) TableName() string {
+	return "operations_gopher_mart"
+}
+
+func (UserDB) TableName() string {
+	return "users_gopher_mart"
+}
+
+type OperationAccrual struct {
+	OrderNumber string  `json:"number"`
+	Status      string  `json:"status"`
+	Points      float64 `json:"accrual,omitempty"`
+	UploadedAt  string  `json:"uploaded_at"`
+	Operation   string  `json:"-"`
+}
+
+type OperationWithdraw struct {
+	OrderNumber string  `json:"order"`
+	Status      string  `json:"-"`
+	Points      float64 `json:"sum,omitempty"`
+	UploadedAt  string  `json:"uploaded_at"`
+	Operation   string  `json:"-"`
+}
+
+const (
+	accrual  = "accrual"
+	withdraw = "withdraw"
+
+	newOrder   = "NEW"
+	processing = "PROCESSING"
+	processed  = "PROCESSED"
+	//invalid    = "INVALID"
+)
+
+type DBI interface {
+	Connect(ctx context.Context, connStr string) (err error)
+	Ping(ctx context.Context) error
+	Close() error
+
+	RegisterUser(ctx context.Context, login string, pass string) (tokenJWT string, err error)
+	LoginUser(ctx context.Context, login string, pass string) (tokenJWT string, err error)
+
+	WriteOrderAccrual(ctx context.Context, order string, user string) (err error)
+	ReadAllOrderAccrualUser(ctx context.Context, user string) (ops []OperationAccrual, err error)
+	ReadUserPoints(ctx context.Context, user string) (u UserPoints, err error)
+	WithdrawnUserPoints(ctx context.Context, user string, order string, sum float64) (err error)
+	WriteOrderWithdrawn(ctx context.Context, order string, user string, point float64) (err error)
+	ReadAllOrderWithdrawnUser(ctx context.Context, user string) (ops []OperationWithdraw, err error)
+
+	ReadAllOrderAccrualNoComplite(ctx context.Context) (orders []orderstruct, err error)
+	UpdateOrderAccrual(ctx context.Context, login string, orderAccrual requestAccrual) (err error)
+}
+
+type Database struct {
+	connection *sql.DB
+}
+
+func InitDB() (*Database, error) {
+	return &Database{}, nil
+}
+
+func (db *Database) Connect(ctx context.Context, connStr string) (err error) {
+	pdb, err := gorm.Open(postgres.Open(connStr), &gorm.Config{})
+	if err != nil {
+		return err
+	}
+	db.connection, err = pdb.DB()
+	if err != nil {
+		return err
+	}
+
+	pdb.AutoMigrate(&OperationDB{})
```

Comment:
```suggestion
	err=pdb.AutoMigrate(&OperationDB{}, &UserDB{})
```
и проверить на ошибку.